### PR TITLE
add cancellation support to more async methods

### PR DESCRIPTION
**Pull request by wpeter-sfdc**
and update UT project to use latest version of sqlite.uap and fix nuget references to use UWP nuget packages

### CODE REVIEW

**Comment by wpeter-sfdc** [comment]
@kchitalia @bhariharan

**Comment by jhsfdc** [comment]
:+1:

**Comment by kchitalia** [comment]
When an existing consumer of the SDK upgrades and calls the methods that have been changed here, will they see errors as a new parameter is passed in a all these methods?

**Comment by wpeter-sfdc** [comment]
@kchitalia It's an optional parameter so no.

However, I did change the name of the method Logout to LogoutAsync to follow .NET async naming conventions... a consumer would get a build error and need to use the new name in their code (change all calls to `Logout` to be `LogoutAsync`).

I can revert the LogoutAsync rename if that's a problem.

**Comment by kchitalia** [comment]
I think we should keep it as LogoutAsync. All the other changes look good too.
:+1:

File: SalesforceSDK/Core/Auth/AccountManager.cs [hunk 1]
```diff
@@ -166,18 +169,20 @@ public static void SwitchAccount()
         /// </summary>
         /// <param name="loginOptions"></param>
         /// <param name="authResponse"></param>
-        public static async Task<Account> CreateNewAccount(LoginOptions loginOptions, AuthResponse authResponse)
+        /// <param name="cancellationToken"></param>
+        public static async Task<Account> CreateNewAccount(LoginOptions loginOptions, AuthResponse authResponse,
```
Comment (jbernoudy) [inline SalesforceSDK/Core/Auth/AccountManager.cs]: CreateNewAccountAsync (unless you are undoing the changes to other methods)

File: SalesforceSDK/Core/Net/HttpCall.cs [hunk 2]
```diff
@@ -274,9 +274,9 @@ public static HttpCall CreatePost(string url, string requestBody)
         /// </summary>
         /// <typeparam name="T"></typeparam>
         /// <returns></returns>
-        public async Task<T> ExecuteAndDeserializeAsync<T>()
```
Comment (jbernoudy) [inline SalesforceSDK/Core/Net/HttpCall.cs]: This comment block should be filled in, and the exception added to it

File: SalesforceSDK/Core/Rest/RestClient.cs [hunk 3]
```diff
@@ -36,7 +36,7 @@ namespace Salesforce.SDK.Rest
 {
     public delegate void AsyncRequestCallback(IRestResponse response);
 
-    public delegate Task<string> AccessTokenProvider();
+    public delegate Task<string> AccessTokenProvider(CancellationToken cancellationToken = default(CancellationToken));
```
Comment (jbernoudy) [inline SalesforceSDK/Core/Rest/RestClient.cs]: This should be documented.

File: SalesforceSDK/Core/Auth/AccountManager.cs (hunk 1 above)
Comment (kchitalia) [inline SalesforceSDK/Core/Auth/AccountManager.cs]: I thought we already renamed all the methods. I think a new pull request for that should be good.

**Comment by jbernoudy** [comment]
Just a note that  SwitchToAccount was renamed to SwitchToAccountAsync so that is another breaking change to watch out for :-)

**Pull request merged**